builtin/logical/aws/path_roles.go

```diff
@@ -80,6 +80,15 @@ will be passed in as the Policy parameter to the AssumeRole or
 GetFederationToken API call, acting as a filter on permissions available.`,
 			},
 
+			"iam_groups": &framework.FieldSchema{
```

Comment:
Question, should we use `iam_group_arns` here instead? The reason I ask is, I notice that other places use policy ARNs and role ARNs instead of their names. My thought is that if we used the names of the groups, someone could create a group named "foo group". Then if someone else deleted "foo group" and remade it, this eventually might not point at the group or permissions originally intended.

Reply:
Good question. I think the reason ARNs are used for policies and roles is because the full ARN is required in the AWS API calls where those parameters are being used:

- [[AttachUserPolicy](https://docs.aws.amazon.com/IAM/latest/APIReference/API_AttachUserPolicy.html)] [[usage](https://github.com/hashicorp/vault/blob/53ada8bd0f15b1326cb28d1882e821c36c49c13f/builtin/logical/aws/secret_access_keys.go#L259-L263)]
- [[AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)] [[usage](https://github.com/hashicorp/vault/blob/53ada8bd0f15b1326cb28d1882e821c36c49c13f/builtin/logical/aws/secret_access_keys.go#L171-L182)]

And the API call where the groups parameter is used only accepts the name: 

- [[AddUserToGroup](https://docs.aws.amazon.com/IAM/latest/APIReference/API_AddUserToGroup.html)] [[usage](https://github.com/hashicorp/vault/blob/53ada8bd0f15b1326cb28d1882e821c36c49c13f/builtin/logical/aws/secret_access_keys.go#L285-L288)]

If we want to require group ARNs we certainly can for consistency, but it also appears that the ARN isn’t a unique identifier. i.e. creating group foo, deleting it, then creating foo again will result in the same ARN: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-arns

Though I did find that there is a [unique ID for each group created](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-unique-ids) that can be retrieved from the API, so perhaps we could store that and check it when new creds are created to verify the IAM group hasn’t been recreated. I think that would add another AWS API call per group during user generation, and it doesn’t look like the `role_arns` and `policy_arns`  go to this length.

Maybe the name `iam_groups` is too ambiguous? Should it be something more like `group_names`? Or just `groups`? (It occurs to me that the other parameters are IAM things, but don’t have `iam` in the name.)

Reply:
Thanks for the explanation, that makes a lot of sense. My thoughts on it are, then, that it would be OK to leave it as it is, but I'd like to gut check that. @kalafut do you agree it should be OK here to use group names instead of ARNs?

Reply:
I'm good with using group names here.